File: webui/src/main/scala/com/taskadapter/webui/config/ConnectorFieldLoader.scala
package com.taskadapter.webui.config

import com.taskadapter.model.{CustomString, Field}
import scala.collection.JavaConverters._

class ConnectorFieldLoader(fields: java.util.List[Field[_]]) {

  def getTypeForFieldName(fieldName: String): Field[_] = {
    fields.asScala.find(f => f.getFieldName().equals(fieldName)).getOrElse(new CustomString(fieldName))
  }
}

File: webui/src/main/java/com/taskadapter/webui/config/ConnectorFieldLoader.java
package com.taskadapter.webui.config;

import com.taskadapter.model.CustomString;
import com.taskadapter.model.Field;

import java.util.List;

public class ConnectorFieldLoader {
    private final List<Field<?>> fields;

    public ConnectorFieldLoader(List<Field<?>> fields) {
        this.fields = fields;
    }

    Field<?> getTypeForFieldName(String fieldName) {
        return fields.stream()
                .filter(f -> f.getFieldName().equals(fieldName))
                .findFirst()
                .orElse(new CustomString(fieldName));
    }
}
